src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantDataController.java

```diff
@@ -0,0 +1,70 @@
+package org.sagebionetworks.bridge.spring.controllers;
+
+import org.sagebionetworks.bridge.dynamodb.DynamoParticipantData;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.ParticipantData;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.StatusMessage;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.services.ParticipantDataService;
+import org.sagebionetworks.bridge.services.ReportService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.sagebionetworks.bridge.BridgeConstants.API_DEFAULT_PAGE_SIZE;
+import static org.sagebionetworks.bridge.BridgeUtils.getIntOrDefault;
+
+@CrossOrigin
+@RestController
+public class ParticipantDataController extends BaseController {
+
+    private ParticipantDataService participantDataService;
+
+    @Autowired
+    final void setReportService(ParticipantDataService participantDataService) {
+        this.participantDataService = participantDataService;
+    }
+
+    @GetMapping("/v4/users/self/configs")
+    public ForwardCursorPagedResourceList<String> getAllDataForUser(@RequestParam(required = false) String offsetKey,
+                                                                    @RequestParam(required = false) String pageSize) {
+        UserSession session = getAuthenticatedSession();
```

Comment:
For consistency, let's also use getAuthenticatedAndConsentedSession() for all the getters. They can't write participant data unless they're consented, and calling the get methods before they do will always return empty results or 404 Not Found.